numpy/doc/structured_arrays.py

```diff
@@ -6,231 +6,508 @@
+like views into the original array, such that modifying the scalar will modify
+the original array. Structured scalars also support access and assignment by
+field name::
+
+ >>> x = np.array([(1,2),(3,4)], dtype=[('foo', 'i8'), ('bar', 'f4')])
+ >>> s = x[0]
+ >>> s['bar'] = 100
+ >>> x
+ array([(1, 100.), (3, 4.)],
+       dtype=[('foo', '<i8'), ('bar', '<f4')])
 
-You can access multiple fields at once using a list of field names: ::
+Similarly to tuples, structured scalars can also be indexed with an integer::
 
- >>> x = np.array([(1.5,2.5,(1.0,2.0)),(3.,4.,(4.,5.)),(1.,3.,(2.,6.))],
-         dtype=[('x','f4'),('y',np.float32),('value','f4',(2,2))])
+ >>> scalar = np.array([(1, 2., 3.)], dtype='i,f,f')[0]
+ >>> scalar[0]
+ 1
+ >>> scalar[1] = 4
 
-Notice that `x` is created with a list of tuples. ::
+Thus, tuples might be though of as the native Python equivalent to numpy's
+structured types, much like native python integers are the equivalent to
+numpy's integer types. Structured scalars may be converted to a tuple by
+calling :func:`ndarray.item`::
 
- >>> x[['x','y']]
- array([(1.5, 2.5), (3.0, 4.0), (1.0, 3.0)],
-      dtype=[('x', '<f4'), ('y', '<f4')])
- >>> x[['x','value']]
- array([(1.5, [[1.0, 2.0], [1.0, 2.0]]), (3.0, [[4.0, 5.0], [4.0, 5.0]]),
-       (1.0, [[2.0, 6.0], [2.0, 6.0]])],
-      dtype=[('x', '<f4'), ('value', '<f4', (2, 2))])
+ >>> scalar.item(), type(scalar.item())
+ ((1, 2.0, 3.0), tuple)
 
-The fields are returned in the order they are asked for.::
+Viewing Structured Arrays Containing Objects
+--------------------------------------------
 
- >>> x[['y','x']]
- array([(2.5, 1.5), (4.0, 3.0), (3.0, 1.0)],
-      dtype=[('y', '<f4'), ('x', '<f4')])
+In order to prevent clobbering of object pointers in fields of
+:class:`numpy.object` type, numpy currently does not allow views of structured
+arrays containing objects.
 
-Filling structured arrays
-=========================
+Structure Comparison
+--------------------
 
-Structured arrays can be filled by field or row by row. ::
+If the dtypes of two structured arrays are equivalent, testing the equality of
+the arrays will result in a boolean array with the dimension of the original
+arrays, with elements set to True where all fields of the corresponding
+structures are equal. Structured dtypes are equivalent if the field names,
+dtypes and titles are the same, ignoring endianness, and the fields are in
+the same order::
 
- >>> arr = np.zeros((5,), dtype=[('var1','f8'),('var2','f8')])
- >>> arr['var1'] = np.arange(5)
+ >>> a = np.zeros(2, dtype=[('a', 'i4'), ('b', 'i4')])
+ >>> b = np.ones(2, dtype=[('a', 'i4'), ('b', 'i4')])
+ >>> a == b
+ array([False, False], dtype=bool)
 
-If you fill it in row by row, it takes a take a tuple
-(but not a list or array!)::
+Currently, if the dtypes of two arrays are not equivalent all comparisons will
+return ``False``. This behavior is deprecated as of numpy 1.10 and may change
```

Comment:
What might be the alternative?

Reply:
Currently in this case we get:
```
[1]: a = np.zeros(3, dtype='f,f,f')
[2]: b = np.zeros(3, dtype='f,f')
[3]: a == b
FutureWarning: elementwise == comparison failed and returning scalar instead; this will raise an error or perform elementwise comparison in the future
```
I'll reword the text to more accurately describe what happens.